homeassistant/components/powerwall/sensor.py

```diff
@@ -281,3 +287,126 @@ def native_value(self) -> float | None:
         if TYPE_CHECKING:
             assert meter is not None
         return meter.get_energy_imported()
+
+
+class PowerWallBatterySensor(BatteryEntity, SensorEntity, Generic[_ValueT]):
+    """Representation of an Powerwall Battery sensor."""
+
+    entity_description: PowerwallSensorEntityDescription[BatteryResponse, _ValueT]
+
+    def __init__(
+        self,
+        powerwall_data: PowerwallRuntimeData,
+        battery: BatteryResponse,
+        description: PowerwallSensorEntityDescription[BatteryResponse, _ValueT],
+    ) -> None:
+        """Initialize the sensor."""
+        self.entity_description = description
+        super().__init__(powerwall_data, battery)
+        self._attr_translation_key = description.translation_key
+        self._attr_unique_id = f"{self.base_unique_id}_{description.key}"
+
+    @property
+    def native_value(self) -> float | int | str:
+        """Get the current value."""
+        return self.entity_description.value_fn(self.battery_data)
+
+
+def _get_battery_charge(battery_data: BatteryResponse) -> float:
+    """Get the current value in %."""
+    ratio = float(battery_data.energy_remaining) / float(battery_data.capacity)
+    return round(100 * ratio, 1)
+
+
+BATTERY_INSTANT_SENSORS: list[PowerwallSensorEntityDescription] = [
+    PowerwallSensorEntityDescription[BatteryResponse, int](
```

Comment:
It looks like we are missing entity category for the ones that should be marked diagnostic. Actually I think all of them should be diagnostic since we likely don't want to flood the default dashboard with a lot of new entities since most people will only need what the gateway provides